rero_ils/modules/patrons/listener.py

```diff
@@ -38,3 +42,30 @@ def enrich_patron_data(sender, json=None, record=None, index=None,
             json['organisation'] = {
                 'pid': org_pid
             }
+
+
+def create_subscription_patron_transaction(sender, record=None, **kwargs):
+    """This method check the patron to know if a subscription is requested.
+
+    This method checks the patron_type linked to a patron and determine if
+    a new subscription patron transaction should be created or not. If a
+    subscriotion is needed, then the 'subscription' patron attribute will be
+    updated.
+    This method should be connect with 'before_record_insert' and
+    'before_record_update'.
+
+    :param record: the record being performed
+    """
+    if record.get('$schema') != get_schema_for_resource(Patron):
+        return
+    if record.patron_type_pid is None:
+        return
+    ptty = PatronType.get_record_by_pid(record.patron_type_pid)
+    if ptty.is_subscription_required and not record.has_valid_subscription:
+        # TODO : (2020-03-27)
+        #   At this time, subscription are only possible for one year. In
+        #   the future, the subscription period should be defined as a
+        #   patron_type attribute.
+        start_date = datetime.now()
```

Comment:
Do we need to consider the time zone?

Reply:
Don't think so. Start/End subscription date are stored as UTC date with format %Y-%m-%d.
In my point of view, timezone isn't relevant here.